pm/sigverifier_test.go

```diff
@@ -0,0 +1,76 @@
+package pm
+
+import (
+	"fmt"
+	"testing"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+func TestVerify(t *testing.T) {
+	msg := []byte("foo")
+	personalMsg := fmt.Sprintf("\x19Ethereum Signed Message:\n%d%s", 32, msg)
+	personalHash := crypto.Keccak256([]byte(personalMsg))
+
+	senderPrivKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sender := crypto.PubkeyToAddress(senderPrivKey.PublicKey)
+
+	senderSig, err := crypto.Sign(personalHash, senderPrivKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	signerPrivKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	signer := crypto.PubkeyToAddress(signerPrivKey.PublicKey)
```

Comment:
Perhaps rename `signer` to `unapprovedSigner` to increase readability? - I noticed we later approve `signer`, but perhaps then we can just create a new signer instance and name that one `approvedSigner`?